code/datums/weather/weather_types/solar_flare.dm

```diff
@@ -0,0 +1,50 @@
+/datum/weather/solar_flare
+	name = "solar flare"
+	desc = "An intense blast of light and heat from the sun, affecting all space around the station."
+
+	telegraph_duration = 30 SECONDS
+	telegraph_message = null // handled via event announcement
+
+	weather_message = "<span class='userdanger'><i>A solar flare has arrived! Find shelter!</i></span>"
+	weather_overlay = "light_ash"
+	weather_duration_lower = 5 MINUTES
+	weather_duration_upper = 15 MINUTES
+	weather_color = COLOR_YELLOW
+	weather_sound = 'sound/misc/bloblarm.ogg' // also used by radiation storm and SM
+
+	end_duration = 10 // wind_down() does not do anything for this event, so we just trigger end() semi-immediately
+	end_message = null
+	area_type = /area/space // read generate_area_list() as well below
+	protected_areas = list()
+	target_trait = STATION_LEVEL
+	immunity_type = "burn"
+
+/datum/weather/solar_flare/generate_area_list()
+	..()
+	var/list/bonus_areas = list()
+	for(var/V in get_areas(/area/solar))
```

Comment:
No need for this loop.
Just a `bonus_areas = get_areas(/area/solar)` will do here